### PR TITLE
refactor(ingestion/tableau): mark the `fetch_size` configuration as deprecated.

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

**Review by mayurinehate — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py [hunk 1]
```diff
@@ -380,7 +380,8 @@ class TableauConfig(
 
     fetch_size: int = Field(
         default=250,
-        description="Specifies the number of records to retrieve in each batch during a query execution.",
+        description="[deprecated] Use page_size instead. Specifies the number of records to retrieve in each batch during a query execution.",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:383]: with this change, fetch_size does not do anything, right?

so we should use `pydantic_field_removed` and make it clear that only page size has an impact

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:383]: @hsheth2 

I haven't found `pydantic_field_removed` in pydantic documentation. Could you please add additional detail here for the same.

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:383]: https://github.com/datahub-project/datahub/blob/8cf778dc9b2bb77f9af1ffd979e65fb283579938/metadata-ingestion/src/datahub/configuration/validate_field_removal.py#L10

Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py:385]: Oh!. Yesterday I searched the code base as per your old comment `pydantic_field_removed `. Got it now

Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py]: done

File: metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py [hunk 2]
```diff
@@ -382,7 +383,7 @@ class TableauConfig(
 
     fetch_size: int = Field(
         default=250,
-        description="[deprecated] Use page_size instead. Specifies the number of records to retrieve in each batch during a query execution.",
+        description="[removed] The 'fetch_size' field is no longer used. Please use 'page_size' instead.",
         hidden_from_docs=True,
     )
```
[the remark this answers is not in the document]
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py]: done

**Review by mayurinehate — APPROVED** [review]

**Comment by hsheth2** [comment]
Dagster failures are unrelated.

**Pull request merged**